python-frontend/src/main/java/org/sonar/python/cfg/fixpoint/UsageVisitor.java

```diff
@@ -78,5 +82,14 @@ public boolean isWrite() {
     public boolean isRead() {
       return isRead;
     }
+
+    public void addUsage(Usage usage) {
+      usages.add(usage);
+    }
+
+    // Returns the list of symbol usages within the visited element
+    public List<Usage> usages() {
```

Comment:
Why a list if we only care about the first element? Can the list be empty?

Reply:
Indeed this use case only required the first element but I felt returning the list might be useful for other cases, and more understandable as well, as discussed with @andrea-guarino-sonarsource 
The list should never be empty.
Do you think I should keep it to the first element only?

Reply:
I don't know enough about this class to have an opinion and I don't think it deserves more discussions.